float32    TransformLength
bool[]     JointMode
float32[]  JointData